pkg/controller/bucket/bucket_controller.go

```diff
@@ -99,18 +100,37 @@ func (bl *bucketListener) Add(ctx context.Context, obj *v1alpha1.Bucket) error {
 	}
 
 	req := osspec.ProvisionerCreateBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
+	}
+	if len(obj.Spec.Parameters["BucketInstanceName"]) > 0 {
```

Comment:
This is a nit that may only be stylistic -- is there a reason not to do the checking of the key checking as follows?
```
	if bktInstanceName, ok := obj.Spec.Parameters["BucketInstanceName"]); ok { ...
        else if bktPrefix, ok := obj.Spec.Parameters["BucketPrefix"]; ok { ...

```